node/core/pvf/src/prepare/worker.rs

```diff
@@ -128,38 +161,46 @@ pub async fn start_work(
 									);
 									Selected::IoErr
 								})
-						} else {
-							Selected::Done(result)
 						}
 					} else {
-						// We received invalid bytes from the worker.
-						let bound_bytes = &response_bytes[..response_bytes.len().min(4)];
-						gum::warn!(
-							target: LOG_TARGET,
-							worker_pid = %pid,
-							"received unexpected response from the prepare worker: {}",
-							HexDisplay::from(&bound_bytes),
-						);
-						Selected::IoErr
+						Selected::Done(result)
 					}
-				},
-				Ok(Err(err)) => {
-					// Communication error within the time limit.
+				} else {
+					// We received invalid bytes from the worker.
+					let bound_bytes = &response_bytes[..response_bytes.len().min(4)];
 					gum::warn!(
 						target: LOG_TARGET,
 						worker_pid = %pid,
-						"failed to recv a prepare response: {:?}",
-						err,
+						"received unexpected response from the prepare worker: {}",
+						HexDisplay::from(&bound_bytes),
 					);
 					Selected::IoErr
-				},
-				Err(_) => {
-					// Timed out.
-					Selected::Deadline
-				},
-			};
+				}
+			},
+			Ok(Err(err)) => {
+				// Communication error within the time limit.
+				gum::warn!(
+					target: LOG_TARGET,
+					worker_pid = %pid,
+					"failed to recv a prepare response: {:?}",
+					err,
+				);
+				Selected::IoErr
+			},
+			Err(_) => {
+				// Timed out.
+				gum::warn!(
+					target: LOG_TARGET,
+					worker_pid = %pid,
+					"did not recv a prepare response within the time limit",
+				);
+				Selected::Deadline
+			},
+		};
 
 		match selected {
+			// Timed out. This should already be logged by the child.
+			Selected::Done(Err(PrepareError::TimedOut)) => Outcome::TimedOut,
```

Comment:
That's true. I didn't think it was necessary to have two different `TimedOut` err types. Can always address as part of https://github.com/paritytech/polkadot/issues/3754 if someone feels otherwise.